event.go

```diff
@@ -67,10 +71,23 @@ func (ed *DefaultEventHandler) Post(height uint64, msg *types.EventData) error {
 	}
 	// TODO: this is stupid, fix it
 	eventData := EventData(*msg)
-	ed.stash.add(height, &eventData)
+	ed.eventCache = append(ed.eventCache, &eventData)
 	return nil
 }
 
+func (ed *DefaultEventHandler) Commit(height uint64) error {
+	for _, eventData := range ed.eventCache {
+		ed.stash.add(height, eventData)
+	}
+	// clear event cache
+	ed.eventCache = make([]*EventData, 0)
```

Comment:
Just do `ed.eventCache = nil` anywhere you want to clear it, no point calling `make` if the capacity is zero.

Reply:
Done